### PR TITLE
fix(#11239): Set fields disabled state based on form and field disabl…

### PR DESCRIPTION
**Pull request by thomasdbock**
Quickfix on #11239 
Needs some love by somebody who knows the internals better than I do.

### CODE REVIEW

File: src/logic/createFormControl.ts [hunk 1]
```diff
@@ -1081,8 +1081,11 @@ export function createFormControl<
       _subjects.state.next({ disabled });
       iterateFieldsByAction(
         _fields,
-        (ref) => {
-          ref.disabled = disabled;
+        (ref, name) => {
+          const currentField = get(_fields, name);
+          if (currentField) {
+            ref.disabled = currentField._f.disabled || disabled;
```
Comment (bluebill1049) [inline src/logic/createFormControl.ts:1087]: probably need to check `currentField._f.disabled` is defined as well.

**Review by bluebill1049 — COMMENTED** [review]
Thanks very much for the PR! could you please include a unit test for this usecase well.

File: src/logic/createFormControl.ts (hunk 1 above)
Comment (thomasdbock) [reply src/logic/createFormControl.ts]: Had a hard time figuring out if there is a standard way you like to check these, hope this is sufficient :) 
Added a test as well.

**Review by bluebill1049 — DISMISSED** [review]
LGTM. 👍

**Review by bluebill1049 — APPROVED** [review]

**Pull request merged**